packages/commerce-sdk-react/src/hooks/ShopperSearch/query.ts

```diff
@@ -19,22 +25,59 @@ the product search hit. The search result contains only products that are online
  * @see {@link https://salesforcecommercecloud.github.io/commerce-sdk-isomorphic/classes/shoppersearch.shoppersearch-1.html#productsearch} for more information on the parameters and returned data type.
  * @returns An object describing the state of the request.
  */
-export const useProductSearch = (
-    arg: Argument<Client['productSearch']>
-): UseQueryResult<DataType<Client['productSearch']>, Error> => {
-    const {shopperSearch: client} = useCommerceApi()
-    return useAsync(['product-search', arg], () => client.productSearch(arg))
+function useProductSearch(
+    arg: Omit<UseProductSearchArg, 'rawResponse'> & {rawResponse?: false},
+    options?: UseQueryOptions<DataType<Client['productSearch']> | Response, Error>
+): UseQueryResult<DataType<Client['productSearch']>, Error>
+function useProductSearch(
+    arg: Omit<UseProductSearchArg, 'rawResponse'> & {rawResponse?: true},
```

Comment:
```suggestion
    arg: Omit<UseProductSearchArg, 'rawResponse'> & {rawResponse: true},
```
Minor: in ShopperProduct, it doesn't have the `?`. I believe the way we think of it is like this:
- First of all, `rawResponse` can be undefined or equivalently `false` → thus, `?` is needed.
- Then `rawResponse` can also be `true` → `?` is not needed here

Is that why there is no `?` when `rawResponse` is true @alexvuong ?

Reply:
Yes that is correct, by default, rawResponse is false, so if it is not presented, it is equivalent to false, which means it can be optional. Meanwhile, if it is true, it has to be set.

Reply:
@shethj i believe the ^ conversation means that we need to remove the `?` on line 33.